core/src/test/java/apoc/trigger/TriggerClusterRoutingTest.java

```diff
@@ -51,87 +57,130 @@ public static void bringDownCluster() {
+        failsInFollowers(installTrigger, triggerName, TRIGGER_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME);
     }
 
     @Test
     public void testTriggerDropAllowedOnlyInSysLeaderMember() {
-        final String query = "CALL apoc.trigger.drop('neo4j', $name)";
-        triggerInSysLeaderMemberCommon(query, TRIGGER_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME);
+        final String installTrigger = "CALL apoc.trigger.install('neo4j', $name, 'RETURN 1', {})";
+        final String dropTrigger = "CALL apoc.trigger.drop('neo4j', $name)";
+        String triggerName = randomTriggerName();
+
+        succeedsInLeader(installTrigger, triggerName, SYSTEM_DATABASE_NAME);
+        succeedsInLeader(dropTrigger, triggerName, SYSTEM_DATABASE_NAME);
+        failsInFollowers(dropTrigger, triggerName, TRIGGER_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME);
     }
 
     @Test
-    public void testTriggerShowAllowedOnlyInSysLeaderMember() {
-        final String query = "CALL apoc.trigger.show('neo4j')";
-        final BiConsumer<Session, String> testTrigger = (session, name) -> testCallEmpty(session, query, Collections.emptyMap());
-        triggerInSysLeaderMemberCommon(query, TRIGGER_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME, true, testTrigger);
+    public void testTriggerShowAllowedInAllSystemInstances() {
+        final String installTrigger = "CALL apoc.trigger.install('neo4j', $name, 'RETURN 1', {})";
+        final String showTrigger = "CALL apoc.trigger.show('neo4j')";
+
+        String triggerName = randomTriggerName();
+        Consumer<Iterator<Map<String, Object>>> checkTriggerIsListed = rows -> {
+            AtomicBoolean showedTrigger = new AtomicBoolean(false);
+            rows.forEachRemaining(row -> {
+                if (row.get("name").equals(triggerName))
+                    showedTrigger.set(true);
+            });
+
+            assertTrue(showedTrigger.get());
+        };
+
+        succeedsInLeader(installTrigger, triggerName, SYSTEM_DATABASE_NAME);
+        succeedsInLeader(showTrigger, triggerName, SYSTEM_DATABASE_NAME, checkTriggerIsListed);
+        succeedsInFollowers(showTrigger, triggerName, SYSTEM_DATABASE_NAME, checkTriggerIsListed);
     }
 
-    private static void triggerInSysLeaderMemberCommon(String query, String triggerNotRoutedError, String dbName) {
-        final BiConsumer<Session, String> testTrigger = (session, name) -> testCall(session, query,
-                Map.of("name", name),
-                row -> assertEquals(name, row.get("name")));
-        triggerInSysLeaderMemberCommon(query, triggerNotRoutedError, dbName, false, testTrigger);
+    private static void succeedsInLeader(String triggerOperation, String triggerName, String dbName) {
+        for (Neo4jContainerExtension instance: clusterMembers) {
+            Session session = getSessionForDb(instance, dbName);
+
+            if (dbIsWriter(SYSTEM_DATABASE_NAME, session, getBoltAddress(instance))) {
```

Comment:
So if this doesn't evaluate to true, we will not actually test the trigger operation succeeded in leader? I assume there will always be a writer though? :)

Reply:
There must always be a leader (=writer) for the system database among the instances of the cluster, so we should always be testing something here